### PR TITLE
URDF documentation Update

### PR DESCRIPTION
**Pull request by vidurvij-Unity**
### Documentation update on primitive sizing
The original issue described the inability of the user to export the size of the primitive he created to the URDF file. He was trying to export the size of the box he created to a URDF file using the export feature but the URDF file always exported the size [1 1 1]. This is an expected behavior because URDF importer does not read or write size of the primitives in the [BoxCollider](https://docs.unity3d.com/ScriptReference/BoxCollider-size.html), CapsuleCollider or the Spherical Collider. Instead it change the size of the visual and collider meshes using the scale of the parent gameObject of the mesh. 

This is done because:
1. To have consistency between the way the size of the mesh is changed between primitive collider meshes and other meshes.
2. This API also works by changing the scale of the mesh under the hood. 
### Adding a list of supported URDF tags
Adding a list of supported URDF tags which are read from a file and attributed to an Articulation Body property in Unity.

### Useful links (GitHub issues, JIRA tickets, forum threads, etc.)

https://jira.unity3d.com/browse/AIRO-623
https://jira.unity3d.com/browse/AIRO-642
https://github.com/Unity-Technologies/URDF-Importer/issues/79

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [x] Documentation update
- [ ] Other (please describe)

## Testing and Verification

Test the bug by importing and exporting R2D2 URDF file which is made up box primitives.

### Test Configuration:
- Unity Version: [e.g. Unity 2020.2.0f1]
- Unity machine OS + version: [e.g. Windows 10]
- ROS machine OS + version: [e.g. Ubuntu 18.04, ROS Noetic]
- ROS–Unity communication: [e.g. Docker]

## Checklist
- [x] Ensured this PR is up-to-date with the `dev` branch
- [x] Created this PR to target the `dev` branch
- [ ] Followed the style guidelines as described in the [Contribution Guidelines](../CONTRIBUTING.md)
- [ ] Added tests that prove my fix is effective or that my feature works
- [x] Updated the documentation as appropriate

## Other comments

### CODE REVIEW

File: tutorials/urdf_importer/urdf_appendix.md [hunk 1]
```diff
@@ -331,3 +331,22 @@ The disable collision tag flags the links that need to be ignored to the URDF pa
...
+```xml
+ <collision>
+      <geometry>
+        <cylinder size=".4 .3 .4" />
+      </geometry>
+    </collision>
+<collision>
+```
+
+The size of the primitive will be set in Unity by changing the scale of gameObject containing UrdfCollision script.
```
Comment (sarah-gibson) [inline tutorials/urdf_importer/urdf_appendix.md:345]: This is passive voice, which is a bit confusing. Who will do the setting? Is it the URDF Importer? If so, here's a suggestion: "The URDF Importer will set the size of the primitive using the "scale" parameter of the GameObject that contains the UrdfCollsion script." [if the meaning is still correct]

File: tutorials/urdf_importer/urdf_appendix.md [hunk 2]
```diff
@@ -331,3 +331,22 @@ The disable collision tag flags the links that need to be ignored to the URDF pa
...
+
+| ![Collision gameObject in hierarchy](images/link_hierarchy.png)  | ![Size of primitive set using scale](images/primitive_scale.png) |
+|:---:|:---:|
+
+This is done, as opposed to using the size API of the primitive collider, to have consistency across different mesh types. The sizing API is only available for primitive mesh colliders and not for complex collider meshes and visual meshes. Thats why we use the parent's scaling to change the shape of the primitive mesh. This scale is exported as the primitive size during URDF export.
```
Comment (sarah-gibson) [inline tutorials/urdf_importer/urdf_appendix.md:350]: Suggestion to improve clarity:
"The reason we set the size via the "scale" of the parent GameObject as opposed to the "size" of the primitive collider is to have consistency across different mesh types; the size API is only available for primitive mesh colliders and not for complex collider meshes and visual meshes, whose size must be changed using their parent GameObject's "scale" parameter."

Comment (sarah-gibson) [reply tutorials/urdf_importer/urdf_appendix.md:350]: I don't understand what the last sentence means - can you clarify?

**Review by sarah-gibson — CHANGES_REQUESTED** [review]

File: tutorials/urdf_importer/urdf_appendix.md (hunk 2 above)
Comment (vidurvij-Unity) [reply tutorials/urdf_importer/urdf_appendix.md]: Updated the language

**Review by sarah-gibson — APPROVED** [review]
Looks good, but since this is getting so long, it would be nice to put a TOC on top!

File: tutorials/urdf_importer/urdf_appendix.md [hunk 3]
```diff
@@ -331,3 +382,24 @@ The disable collision tag flags the links that need to be ignored to the URDF pa
 
 Note: You can also manually ignore collisions in Unity using [APIs](https://docs.unity3d.com/ScriptReference/Physics.IgnoreCollision.html).
 
+## Sizing of Primitives
+
+```xml
+ <collision>
```
Comment (hyounesy) [inline tutorials/urdf_importer/urdf_appendix.md:388]: nit: remove extra space.

File: tutorials/urdf_importer/urdf_appendix.md [hunk 4]
```diff
@@ -331,3 +382,24 @@ The disable collision tag flags the links that need to be ignored to the URDF pa
...
+    </collision>
+<collision>
+```
+
+The URDF Importer will set the size of the primitive using the "scale" parameter of the GameObject that contains the UrdfCollsion script.
+
+| ![Collision gameObject in hierarchy](images/link_hierarchy.png)  | ![Size of primitive set using scale](images/primitive_scale.png) |
+|:---:|:---:|
+
+The reason we set the size via the "scale" of the parent GameObject as opposed to the "size" of the primitive collider is to have consistency across different mesh types; the size API is only available for primitive mesh colliders and not for complex collider meshes and visual meshes, whose size must be changed using their parent GameObject's "scale" parameter. The "scale" parameter of the gameObject is also used set the values of primitive "size" in the URDF when performing an URDF export.
```
Comment (hyounesy) [inline tutorials/urdf_importer/urdf_appendix.md:401]: - ... is also used [to] set the values of [the] primitive['s] ...
- ... performing [a] URDF export.

File: tutorials/urdf_importer/urdf_appendix.md [hunk 5]
```diff
@@ -331,3 +382,24 @@ The disable collision tag flags the links that need to be ignored to the URDF pa
...
 
+## Sizing of Primitives
+
+```xml
+ <collision>
+      <geometry>
+        <cylinder size=".4 .3 .4" />
+      </geometry>
+    </collision>
+<collision>
```
Comment (hyounesy) [inline tutorials/urdf_importer/urdf_appendix.md:393]: remove the extra <collision> tag at the bottom and reformat:
``` xml
<collision>
  <geometry>
    <cylinder size=".4 .3 .4" />
  </geometry>
</collision>
```

File: tutorials/urdf_importer/urdf_appendix.md [hunk 6]
```diff
@@ -316,8 +316,59 @@ To address this predicament we have integrated another algorithm to create Conve
...
+URDF defines the robot visually using Visual Meshes, and its collision using Collision Meshes. Collision meshes define the physical volume of the links, and are used to calculate the inertia of the links and also to detect collisions between different physical objects. In Unity, RigidBodies cannot have concave collision meshes, so when importing a concave collision mesh, all concave regions are closed over to produce a convex outline. As a result, the convex shapes might intersect with each other, creating a hindrance in robot movement. To remedy this, we support a ```disable collision``` tag in URDF. To add an exception for collision detection in Unity:
```
Comment (hyounesy) [inline tutorials/urdf_importer/urdf_appendix.md:371]: nit:  Rigidbody
Perhaps you can also link it to the documentation. e.g.: 
In Unity, [Rigidbody](https://docs.unity3d.com/ScriptReference/Rigidbody.html) components cannot ...

**Review by hyounesy — CHANGES_REQUESTED** [review]
- The example xml needs to be corrected.
- Also some minor typos.

File: tutorials/urdf_importer/urdf_appendix.md [hunk 7]
```diff
@@ -316,8 +327,59 @@ To address this predicament we have integrated another algorithm to create Conve
...
+URDF defines the robot visually using Visual Meshes, and its collision using Collision Meshes. Collision meshes define the physical volume of the links, and are used to calculate the inertia of the links and also to detect collisions between different physical objects. In Unity, [RigidBodies](https://docs.unity3d.com/ScriptReference/Rigidbody.html) cannot have concave collision meshes, so when importing a concave collision mesh, all concave regions are closed over to produce a convex outline. As a result, the convex shapes might intersect with each other, creating a hindrance in robot movement. To remedy this, we support a ```disable collision``` tag in URDF. To add an exception for collision detection in Unity:
```
Comment (hyounesy) [inline tutorials/urdf_importer/urdf_appendix.md:382]: consistency nit: Rigid~~B~~[b]odies, or better [Rigidbody components]

**Review by hyounesy — APPROVED** [review]

**Pull request merged**